packages/legacy-client/src/client/convertWireToOsdkObject.ts

```diff
@@ -119,5 +138,133 @@ export function convertWireToOsdkObject<
   }
 
   Object.setPrototypeOf(obj, proto);
+  setPropertyPrototypes<T, O>(client, apiName, obj);
+
   return obj as OsdkLegacyObjectFrom<O, T>;
 }
+
+function setPropertyPrototypes<
+  T extends ObjectTypesFrom<O> & string,
+  O extends OntologyDefinition<any>,
+>(client: ThinClient<O>, apiName: T, obj: OntologyObjectV2) {
+  for (
+    const [k, v] of Object.entries(client.ontology.objects[apiName].properties)
+  ) {
+    if (!obj[k]) {
+      continue;
+    }
+    switch (v.type) {
+      case "attachment":
+        createPropertyPrototype(
+          obj,
+          k,
+          (value) => AttachmentProperty(client, value.rid),
+        );
+        break;
+      case "geopoint":
+        createPropertyPrototype(
+          obj,
+          k,
+          (value) => GeoPoint.fromGeoJson(value),
+        );
+        break;
+      case "geoshape":
+        createPropertyPrototype(
+          obj,
+          k,
+          (value) => GeoShape.fromGeoJson(value),
+        );
+        break;
+      case "datetime":
+        createPropertyPrototype(
+          obj,
+          k,
+          (value) => LocalDate.fromISOString(value),
+        );
+        break;
+      case "timestamp":
+        createPropertyPrototype(
+          obj,
+          k,
+          (value) => Timestamp.fromISOString(value),
+        );
+        break;
+      case "string":
+      case "boolean":
+      case "double":
+      case "integer":
+      case "short":
+      case "long":
+      case "float":
+      case "decimal":
+      case "byte":
+        break;
+      default:
+        const _: never = v.type;
+    }
+  }
+}
+
+function createPropertyPrototype<T>(
+  obj: OntologyObjectV2,
+  k: string,
+  constructor: (value: any) => T,
+) {
+  if (!obj[k]) {
```

Comment:
This is a *little* sketchy because of things that can be falsy like `0` and `false`, but they don't get this far.